### PR TITLE
New hook on UpdatePlayerMovement() to modify CockpitPositionTransformed()

### PR DESCRIPTION
**Pull request by morallo**
This finally gets rid of the jitter.

### CODE REVIEW

File: cockpitlook.cpp [hunk 1]
```diff
@@ -1364,7 +1365,10 @@ int CockpitLookHook(int* params)
 				if (g_bInHyperspace)
 					g_headPos = g_prevHeadingMatrix * g_headPos;
```
Comment (Prof-Butts) [inline cockpitlook.cpp:1366]: We need to check what happens with the head tracking while in hyperspace.

File: cockpitlook.cpp [hunk 2]
```diff
@@ -1364,7 +1365,10 @@ int CockpitLookHook(int* params)
 				if (g_bInHyperspace)
 					g_headPos = g_prevHeadingMatrix * g_headPos;
 				else
-					g_headPos = HeadingMatrix * g_headPos;
+					1;
```
Comment (Prof-Butts) [inline cockpitlook.cpp:1368]: If the else is not used, it's probably better to just remove it.

File: cockpitlook.cpp [hunk 3]
```diff
@@ -1838,25 +1842,26 @@ int PlayerCameraUpdateHook(int* params)
...
+{
+	//log_debug("Running hooked Vector3Transform in CockpitPositionTransformHook\n");
+	
+	Vector3* vec = (Vector3*)params[0];	
+	
+	// Recommended value for g_fXWAUnitsToMetersScale = 25.
+	vec->x += (g_fXWAUnitsToMetersScale * g_headPos.x);
+	vec->z += (g_fXWAUnitsToMetersScale * g_headPos.y);
+	vec->y -= (g_fXWAUnitsToMetersScale * g_headPos.z);
+	void* (*Vector3Transform)(void*, void*) = (void* (*)(void*, void*)) 0x439B30;
```
Comment (Prof-Butts) [inline cockpitlook.cpp:1860]: This looks like it can be moved outside this function and made a global function pointer.

**Review by Prof-Butts — APPROVED** [review]

**Comment by Prof-Butts** [comment]
Great job! We still need to polish a few bits, but this looks great! Thanks for your help!

File: cockpitlook.cpp (hunk 1 above)
Comment (morallo) [reply cockpitlook.cpp:1366]: From what I tested it works

**Pull request merged**